packages/types/src/documents/app/row.ts

```diff
@@ -1,22 +1,76 @@
+  // as a toggle or checkbox within forms/grids.
   BOOLEAN = "boolean",
+  // a JSON type, this type is always an array of strings, called Multi-select within Budibase.
+  // This type can be compared to the options type, as it functions similarly, but allows picking
+  // multiple options rather than a single option.
   ARRAY = "array",
+  // a string type, this is always a string when input/returned from the API, called Date/Time within
+  // Budibase. We utilise ISO date strings for representing dates, this type has a range of sub-types
+  // to restrict it to date only, time only and ignore timezone capabilities.
   DATETIME = "datetime",
+  // a JSON type, an array of metadata about files held in object storage, called Attachment List within
+  // Budibase. To utilise this type there is an API for uploading files to Budibase, which returns metadata
+  // that can be stored against columns of this type. Currently this is not supported on external databases.
   ATTACHMENTS = "attachment",
+  // a JSON type, similar to the attachments type, called Attachment within Budibase. This type functions
+  // much the same as the attachment list, but only holds a single attachment metadata as an object.
+  // This simpifies the binding experience of using this column type.
   ATTACHMENT_SINGLE = "attachment_single",
+  // a complex type, called Relationships within Budibase. This is the most complex type of Budibase,
+  // nothing should be stored against rows under link columns; this type simply represents the
+  // relationship between tables as part of the table schema. When rows are input to the Budibase API
+  // relationships to be made are represented as a list of row IDs to link. When rows are returned
+  // from the Budibase API it will contain a list of row IDs and display column values of the related rows.
   LINK = "link",
+  // a complex type, called Formulas within Budibase. This type has two variants, static and dynamic, with
+  // static only being supported against internal tables. Dynamic formulas calculate a provided HBS/JS binding
+  // based on the row context and enrich it when rows are being returned from the API. Static bindings calculate
+  // this when rows are being stored, so that the formula output can be searched upon within the DB.
   FORMULA = "formula",
+  // a complex type, called Auto Column within Budibase. This type has a few variants, with options such as a
+  // date for created at/updated at, an auto ID column with auto-increments as rows are saved and a user
+  // relationship type which stores the created by/updated by user details. This sub-types all depend on the
+  // date, number of link types respectively.
   AUTO = "auto",
+  // a JSON type, called JSON within Budibase. This type allows any arbitrary JSON to be input to this column
+  // type, which will be represented a string in the row. This type depends on a schema being provided to make the
```

Comment:
It's not always represented as a string in the row. For external datasources that support it (e.g. Postgres) we use the native JSON column type.